src/plugins/intel_cpu/src/nodes/kernels/jit_kernel_base.hpp

```diff
@@ -0,0 +1,216 @@
+    const size_t ymm_len = 32;
+    const size_t zmm_len = 64;
+
+    explicit jit_kernel_base(x64::cpu_isa_t max_cpu_isa)
+            : x64::jit_generator{nullptr, 256 * 1024, true, max_cpu_isa}
+            , max_cpu_isa_{max_cpu_isa} {}
+    virtual ~jit_kernel_base() {}
+
+    void generate() override {
+        this->preamble();
+        registers_pool_ = RegistersPool::create(max_cpu_isa_);
+        stack_allocator_ = std::make_shared<StackAllocator>(*this);
+        generate(registers_pool_, stack_allocator_);
+        registers_pool_.reset();
+        stack_allocator_.reset();
+        this->postamble();
+    }
+
+    virtual void generate(RegistersPool::Ptr registers_pool, StackAllocator::Ptr stack_allocator) {}
+
+    inline bool is_valid_isa(x64::cpu_isa_t isa) {
+        return x64::is_subset(isa, max_cpu_isa_) && x64::mayiuse(isa);
+    }
+
+    inline void uni_vgatherdps(const Xbyak::Xmm &xmm_val,
+                               const Xbyak::Reg64 &reg_addr,
+                               const Xbyak::Xmm &xmm_index,
+                               const int &scale,
+                               const int &disp,
+                               const Xbyak::Reg &reg_mask) {
+        const size_t kDataTypeSize = sizeof(float);
+        if (is_valid_isa(x64::avx512_core)) {
+            assert(reg_mask.isOPMASK());
+            vgatherdps(xmm_val, ptr[reg_addr + xmm_index * scale + disp]);
+        } else if (is_valid_isa(x64::avx2)) {
+            assert(reg_mask.isYMM());
+            Xbyak::Ymm ymm_mask{reg_mask.getIdx()};
+            vgatherdps(xmm_val, ptr[reg_addr + xmm_index * scale + disp], ymm_mask);
+        } else {
+            const size_t kSimdWidth = x64::cpu_isa_traits<x64::sse41>::vlen / kDataTypeSize;
+            assert(reg_mask.isXMM());
+            Xbyak::Xmm xmm_mask{reg_mask.getIdx()};
+            assert(xmm_val.getKind() == xmm_index.getKind());
+            assert(xmm_index.getKind() == xmm_mask.getKind());
+
+            std::vector<Xbyak::Reg> not_available_reg{reg_addr};
+            const Xbyak::Reg64 idx = registers_pool_->getInplaceFree<Xbyak::Reg64>(not_available_reg);
+            const Xbyak::Reg64 mask = registers_pool_->getInplaceFree<Xbyak::Reg64>(not_available_reg);
+
+            push(idx);
+            push(mask);
+            xor_(idx, idx);
+            xor_(mask, mask);
+
+            for (int i = 0; i < static_cast<int>(kSimdWidth); i++) {
+                Xbyak::Label gather_end;
+                uni_vpextrd(mask, xmm_mask, i);
+                cmp(mask, 0xFFFFFFFF);
+                jne(gather_end, T_NEAR);
+                uni_vpextrd(idx, xmm_index, i);
+                Xbyak::Address addr = ptr[reg_addr + idx * scale + disp];
+                uni_vpinsrd(xmm_val, xmm_val, addr, i);
+                L(gather_end);
+            }
+
+            pop(mask);
+            pop(idx);
+        }
+    }
+
+    inline void uni_vscatterdps(const Xbyak::Reg64& reg_addr,
```

Comment:
The same remarks as for the gather.

Also please move both of this function into PR where they are used.

Reply:
uni_vgatherdps and uni_vscatterdps were excluded from this PR as you requested.